Lattice Strain Postprocessing Scripts

rjk5987 opened the pull request:
Hi @rcarson3 ,

Here are the three scripts for the lattice strain calculation workflow (hopefully I'm doing this pull request correctly). 

1. adios2_extraction.py is an approach to pulling out a list of variables from ADIOS2 outputs made with large meshes in mind, so not the fastest but very careful in its use of memory. Supports many different variables and saves them in ASCII format by variable and cycle.
2. strain_Xtal_to_Sample.py converts the elastic strain tensors for each element from the crystal frame to the sample frame.
3. calc_lattice_strain.py calculates lattice strains as volume-averages of projected strain tensors over elements in crystallographic fibers.

rjk5987 committed:
Add files via upload

rcarson3 commented on scripts/postprocessing/calc_lattice_strain.py:
```diff
@@ -0,0 +1,165 @@
+
+#%% Establish filepaths and directories.
+
+# requires a materials.h5 file created in HEXRD
+mat_file = '/dir/to/materials.h5' #!!!
+# material of interest in material file
+mat = 'IN625' #!!!
```
@rjk5987 potentially within a comment in this file could you show briefly how to create one of those materials files using the HEXRD api or if there is an example in the HEXRD repo point to it here? It would just make it easier for those unfamiliar with HEXRD to be able to get the bare minimum required to run these scripts aka people like me.

rcarson3 reviewed (commented):
@rjk5987 overall it looks fine. I just had one small comment to help future users utilize these scripts y'all have created.

rjk5987 committed:
Update calc_lattice_strain.py

Updated script with straightforward function that bypasses need for materials.h5 file. Much more user-friendly.

rjk5987 committed:
Update calc_lattice_strain.py

Edited description of mat_name argument for make_matl() for clarity.

rjk5987 commented:
@rcarson3 Thanks for pointing that out - a proper materials.h5 file is way overkill for this script. I've commented that bit out to keep it there for those that are familiar with the HEXRD api, and added in its place a straightforward, self-contained function to build a material object from a space group and lattice parameters.

rcarson3 reviewed (approved):
Thanks for putting these together @rjk5987 these are definitely a great addition!
